src/Core/ImageMap.cs

```diff
@@ -241,7 +242,8 @@ public void RemoveItem(Address addr)
             // Merge with next item
             ImageMapItem nextItem;
             if (items.TryGetUpperBound((addr + 1), out nextItem) &&
-                nextItem.DataType is UnknownType)
+                nextItem.DataType is UnknownType &&
+                mergedItem.EndAddress.Equals(nextItem.Address))
             {
```

Comment:
The "==" operator is overloaded for `Address`, so you should feel OK to use it..